src/Compilers/Core/Portable/DiagnosticAnalyzer/IAnalyzerAssemblyResolver.cs

```diff
@@ -2,21 +2,30 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#if NET
+
 using System.Reflection;
+using System.Runtime.Loader;
 
 namespace Microsoft.CodeAnalysis
 {
     /// <summary>
-    /// Allows a host to override how assembly resolution is performed by the <see cref="AnalyzerAssemblyLoader"/>.
+    /// This interface allows hosts to control exactly how a given <see cref="AssemblyName"/> is resolved to an
+    /// <see cref="Assembly"/> instance. This is useful for hosts that need to load assemblies in a custom way like
+    /// Razor or stream based loading.
     /// </summary>
     internal interface IAnalyzerAssemblyResolver
     {
         /// <summary>
-        /// Attempts to resolve an assembly by name.
+        /// Resolve a <see cref="Assembly"/> for the given parameters.
         /// </summary>
-        /// <param name="assemblyName">The assembly to resolve.</param>
-        /// <param name="rootDirectory">The root directory the calling <see cref="AnalyzerAssemblyLoader"/> is configured with.</param>
-        /// <returns>The resolved assembly, or <see langword="null"/></returns>
-        Assembly? ResolveAssembly(AssemblyName assemblyName, string rootDirectory);
+        /// <param name="loader">The <see cref="AnalyzerAssemblyLoader"/> instance that is performing the load</param>
+        /// <param name="directoryContext">The <see cref="AssemblyLoadContext"/> for the current directory</param>
+        /// <param name="assemblyName">The name of the assembly to be loaded</param>
+        /// <param name="directory">The directory where the assembly is being loaded from</param>
+        /// <returns></returns>
+        Assembly? Resolve(AnalyzerAssemblyLoader loader, AssemblyName assemblyName, AssemblyLoadContext directoryContext, string directory);
```

Comment:
don't quite have a good undestanding of having both a directory and a directory context.  i need to figure out the intuition around what this all means.

Consider remarks (or summary) that gives an example of what might be tryignt o be resolved, and what data might flow in all these parameters to ground this.

Reply:
Added some context in remarks